Sources/Parameters.swift

```diff
@@ -0,0 +1,59 @@
+//
+// StencilSwiftKit
+// Copyright (c) 2017 SwiftGen
+// MIT Licence
+//
+
+import Foundation
+
+public enum ParametersError: Error {
+	case invalidSyntax(value: String)
+	case invalidKey(key: String, value: String)
+}
+
+public enum Parameters {
+	typealias Parameter = (key: String, value: String)
+	public typealias StringDict = [String: Any]
+	
+	public static func parse(items: [String]) throws -> StringDict {
+		let parameters: [Parameter] = try items.map { item in
+			let parts = item.components(separatedBy: "=")
+			guard parts.count == 2 else { throw ParametersError.invalidSyntax(value: item) }
+			return (key: parts[0], value: parts[1])
+		}
+		
+		var result = StringDict()
+		for parameter in parameters {
+			result = try parse(item: parameter, result: result)
+		}
+		
+		return result
+	}
+	
+	private static func parse(item: Parameter, result: StringDict) throws -> StringDict {
```

Comment:
Maybe rename that private function to `splitAndRepeat` or similar? Making it have the same name as the public one was confusing about its intend on first read